ZuluContent/Engines/Harvest/Core/HarvestSystem.cs

```diff
@@ -240,17 +241,18 @@ public virtual bool BeginHarvesting(Mobile from, Item tool)
             await Timer.Pause(1000);
 
             if (type != null)
-                def.BonusEffect(from, tool);
+                def.BonusEffect?.Invoke(from, tool);
 
             OnHarvestFinished(from, tool, def, bank, toHarvest);
         }
 
-        public virtual void OnHarvestFinished(Mobile from, Item tool, HarvestDefinition def,
+        public virtual async void OnHarvestFinished(Mobile from, Item tool, HarvestDefinition def,
             HarvestBank bank, object harvested)
         {
             // Loop continuously until the player moves or their tool breaks
-            Timer.DelayCall(TimeSpan.FromSeconds(6),
-                delegate { StartHarvesting(from, tool, harvested); });
+            await Timer.Pause(6000);
```

Comment:
Might be worth making this configurable, i.e. `AutoLoopDelay`

Reply:
Indeed